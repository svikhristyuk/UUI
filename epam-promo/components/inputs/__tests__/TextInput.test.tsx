import React from 'react';
import { TextInput } from '../TextInput';
import renderer from 'react-test-renderer';
import { ReactComponent as CalendarIcon } from '../../../icons/calendar-18.svg';

describe('TextInput', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<TextInput
                value={ null }
                onValueChange={ jest.fn }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<TextInput
                value={ null }
                onValueChange={ jest.fn }
                onAccept={ jest.fn }
                onCancel={ jest.fn }
                icon={ CalendarIcon }
                iconPosition='right'
                size='36'
                isDropdown
                isOpen
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});


