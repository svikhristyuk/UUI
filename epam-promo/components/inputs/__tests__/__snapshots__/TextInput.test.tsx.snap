// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should be rendered correctly 1`] = `
<div
  className="container uui-input-box -clickable root size-36 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`TextInput should be rendered correctly 2`] = `
<div
  className="container uui-input-box -clickable root size-36 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled -clickable"
    style={Object {}}
  >
    <SvgrURL
      className="flipY"
    />
  </div>
</div>
`;
